### PR TITLE
New 'VersionInfoOriginalFileName' directive for the [Setup] section

### PR DESCRIPTION
**Pull request by kleuter**
Sometimes this parameter can be useful for a setup program to give more information to a user/antivirus companies.

### CODE REVIEW

**Comment by kleuter** [comment]
Alternatively, OriginalFileName resource info tag can be set based on OutputBaseFilename

File: Projects/CompResUpdate.pas [hunk 1]
```diff
@@ -42,7 +42,7 @@ procedure ErrorWithLastError(const Msg: String);
 procedure UpdateVersionInfo(const F: TFile;
   const NewBinaryFileVersion, NewBinaryProductVersion: TFileVersionNumbers;
   const NewCompanyName, NewFileDescription, NewTextFileVersion, NewLegalCopyright,
-  NewProductName, NewTextProductVersion: String);
+  NewProductName, NewTextProductVersion, OriginalFileName: String);
```
Comment (martijnlaan) [inline Projects/CompResUpdate.pas]: Can you change this to `NewOriginalFileName`?

**Comment by kleuter** [comment]
Thanks for accepting, Martijn, done. I don't know Delphi, otherwise, I'd also implemented setting this parameter automatically (recommended), probably via VersionInfoSetOriginalFileName (yes/no)

File: Projects/CompResUpdate.pas [hunk 2]
```diff
@@ -42,7 +42,7 @@ procedure ErrorWithLastError(const Msg: String);
 procedure UpdateVersionInfo(const F: TFile;
   const NewBinaryFileVersion, NewBinaryProductVersion: TFileVersionNumbers;
   const NewCompanyName, NewFileDescription, NewTextFileVersion, NewLegalCopyright,
-  NewProductName, NewTextProductVersion, OriginalFileName: String);
+  NewProductName, NewTextProductVersion, NewOriginalFileName: String);
```
Comment (martijnlaan) [inline Projects/CompResUpdate.pas]: You forgot to update the line where the parameter is actually used.

File: whatsnew.htm [hunk 3]
```diff
@@ -33,6 +33,7 @@
 <ul>
    <li>Added new predefined variable <tt>NewLine</tt>.</li>
    <li>Added new <tt>IsWin64</tt>, <tt>SameStr</tt> and <tt>SameText</tt> support functions.</li>
+   <li>Added new <tt>VersionInfoOriginalFileName</tt> directive to the [Setup] section.</li>
```
Comment (martijnlaan) [inline whatsnew.htm]: Should not be under ISPP. See notes of earlier versions on how to list a new directive.

File: ISHelp/isetup.xml [hunk 4]
```diff
@@ -929,6 +929,7 @@ DefaultGroupName=My Program
 <li><link topic="setup_versioninfoproductversion">VersionInfoProductVersion</link></li>
 <li><link topic="setup_versioninfotextversion">VersionInfoTextVersion</link></li>
 <li><link topic="setup_versioninfoversion">VersionInfoVersion</link></li>
+<li><link topic="setup_versioninfooriginalfilename">VersionInfoOriginalFileName</link></li>
```
Comment (martijnlaan) [inline ISHelp/isetup.xml]: Should be moved up a bit so things are still alphabetical.

**Comment by martijnlaan** [comment]
Thanks, I got 3 more comments about your changes, see above. Let me know if you can handle them or not.

**Comment by kleuter** [comment]
Done, thanks for corrections

**Comment by martijnlaan** [comment]
Merged it to a branch here and did some additional tweaks and a fix, see 277c392. Just need to test it.

**Comment by kleuter** [comment]
Thank you! looking forward to 5.6.2 with the new directive

**Pull request merged**